src/library/scala/collection/StrictOptimizedIterableOps.scala

```diff
@@ -237,4 +237,10 @@ trait StrictOptimizedIterableOps[+A, +CC[_], +C]
     (l.result(), r.result())
   }
 
+  // Optimization avoids creation of second collection
+  override def tapEach[U](f: A => U): self.type  = {
+    foreach(f)
+    this
```

Comment:
I think it would be simpler to return `coll` here (and keep using `C` as the return type).